pkg/server/v3/delta_test.go

```diff
@@ -457,3 +458,95 @@ func TestDeltaCallbackError(t *testing.T) {
 		})
 	}
 }
+
+// BENCHMARKS =====================================================================================================
+
+func BenchmarkDeltaResponseHandlers(b *testing.B) {
+	for _, typ := range testTypes {
+		b.Run(typ, func(b *testing.B) {
+			for n := 0; n < b.N; n++ {
+				config := makeMockConfigWatcher()
+				config.deltaResponses = makeDeltaResponses()
+				s := server.NewServer(context.Background(), config, server.CallbackFuncs{})
+
+				resp := makeMockDeltaStream(&testing.T{})
+				// This is a wildcard request since we don't specify a list of resource subscriptions
+				res, err := config.deltaResponses[typ][0].GetDeltaDiscoveryResponse()
+				if err != nil {
+					b.Error(err)
+				}
+				// We only subscribe to one resource to see if we get the appropriate number of resources back
+				resp.recv <- &discovery.DeltaDiscoveryRequest{Node: node, TypeUrl: typ, ResourceNamesSubscribe: []string{res.Resources[0].Name}}
+
+				go func() {
+					err := process(typ, resp, s)
+					assert.NoError(&testing.T{}, err)
+				}()
+
+				select {
+				case res := <-resp.sent:
+					close(resp.recv)
+
+					assert.Equal(&testing.T{}, 1, config.deltaCounts[typ])
+					assert.Empty(&testing.T{}, res.GetSystemVersionInfo())
+				case <-time.After(1 * time.Second):
+					b.Fatalf("got no response")
+				}
+			}
+		})
+	}
+}
+
+func BenchmarkDeltaAggregatedHandlers(b *testing.B) {
```

Comment:
This also needs a b.N loop

Reply:
So I had that in there originally and the benchmark executed the same as not having it. Also the compiler noted this:
```
variable in loop condition never changes
```